velox/functions/prestosql/types/JsonType.cpp

```diff
@@ -640,10 +640,13 @@ struct CastFromJsonTypedImpl {
   // class.
   template <TypeKind kind, typename Dummy = void>
   struct KindDispatcher {
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wreturn-type"
     static simdjson::error_code apply(Input, exec::GenericWriter&) {
```

Comment:
I haven't tried this before, but could we get away with declaring such functions as "[[noreturn]]" instead?

Reply:
Actually [[[noreturn]]](https://en.cppreference.com/w/cpp/language/attributes/noreturn) is a bit confusing, it's not as you would expect the opposite of `[[nodiscard]]` but rather about the return of control flow :face_in_clouds:

Reply:
Can we maybe add a dummy return statement with a comment instead of #pragma's.

```
   return simdjson::UNEXPECTED_ERROR; // Make compiler happy.
```

Reply:
Done, thank you!